src/gui/loginrequireddialog/loginrequireddialog.cpp

```diff
@@ -41,8 +41,16 @@ LoginRequiredDialog::LoginRequiredDialog(AbstractLoginWidget *contentWidget, QWi
 
     // using a stacked widget appears to work better than a plain widget
     // we do this in the setup wizard as well
-    _ui->contentWidget->addWidget(contentWidget);
-    _ui->contentWidget->setCurrentWidget(contentWidget);
+    _ui->contentWidget->setCurrentWidget([this, mode]() -> QWidget * {
```

Comment:
bit late but you can use the enum as index for a stack widget.

Reply:
Hmm, how would I make sure the stacked widget index and the enum are in sync?